src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs

```diff
@@ -0,0 +1,1020 @@
+                }
+                else
+                {
+                    return new ReferenceMap(data);
+                }
+            }
+
+            private static readonly ConditionalWeakTable<ObjectData, ReferenceMap> s_baseDataMap
+                = new ConditionalWeakTable<ObjectData, ReferenceMap>();
+
+            private static ReferenceMap GetBaseReaderData(ObjectData data)
+            {
+                ReferenceMap baseData;
+                if (!s_baseDataMap.TryGetValue(data, out baseData))
+                {
+                    baseData = s_baseDataMap.GetValue(data, _data => new ReferenceMap(_data));
+                }
+
+                return baseData;
+            }
+
+            public int GetNextReferenceId()
+            {
+                _values.Add(null);
+                return _baseDataCount + _values.Count - 1;
+            }
+
+            public void AddValue(int referenceId, object value)
+            {
+                _values[referenceId - _baseDataCount] = value;
+            }
+
+            public object GetValue(int referenceId)
+            {
+                if (_baseData != null)
+                {
+                    if (referenceId < _baseDataCount)
+                    {
+                        return _baseData.GetValue(referenceId);
+                    }
+                    else
+                    {
+                        return _values[referenceId - _baseDataCount];
+                    }
+                }
+                else
+                {
+                    return _values[referenceId];
+                }
+            }
+        }
+
+        internal uint ReadCompressedUInt()
+        {
+            var info = _reader.ReadByte();
+            byte marker = (byte)(info & StreamObjectWriter.ByteMarkerMask);
+            byte byte0 = (byte)(info & ~StreamObjectWriter.ByteMarkerMask);
+
+            if (marker == StreamObjectWriter.Byte1Marker)
+            {
+                return byte0;
+            }
+
+            if (marker == StreamObjectWriter.Byte2Marker)
+            {
+                var byte1 = _reader.ReadByte();
+                return (((uint)byte0) << 8) | byte1;
+            }
+
+            if (marker == StreamObjectWriter.Byte4Marker)
+            {
+                var byte1 = _reader.ReadByte();
+                var byte2 = _reader.ReadByte();
+                var byte3 = _reader.ReadByte();
+
+                return (((uint)byte0) << 24) | (((uint)byte1) << 16) | (((uint)byte2) << 8) | byte3;
+            }
+
+            throw ExceptionUtilities.UnexpectedValue(marker);
+        }
+
+        private string ReadStringValue()
+        {
+            var kind = (EncodingKind)_reader.ReadByte();
+            return kind == EncodingKind.Null ? null : ReadStringValue(kind);
+        }
+
+        private string ReadStringValue(EncodingKind kind)
+        {
+            switch (kind)
+            {
+                case EncodingKind.StringRef_B:
```

Comment:
Why 'Won't fix'.  Names should not be needlessly cryptic.  It doesn't buy us anything and it makes reading the code more difficult #Fix